graql/language/rule-validation.feature

```diff
@@ -702,3 +702,60 @@ Feature: Graql Rule Validation
 
     """
     Then the integrity is validated
+
+
+#CHECK NO DOWN-CASTING AND SIDE-CASTING IS NOT ALLOWED
+  Scenario: when a rule has a down-cast in a rule conclusion, an error is thrown
+    When graql define throws
+      """
+      define
+
+      man sub person;
+
+      rule all-bobs-are-men:
+      when {
+        $p isa person;
+        $p has name 'bob';
+      } then {
+        $p isa man;
+      };
+      """
+    Then the integrity is validated
+
+
+    Scenario: when a rule has a side-cast in a rule, an error is thrown
+      When graql define throws
+      """
+      define
+
+      man sub person;
+      boy sub person;
+
+      rule male-children-are-boys:
+      when {
+        $p isa person;
+        $p has age $a;
```

Comment:
This may throw for the wrong reason - `age` is not defined in the knowledge graph. It should be added, either in this test or in the scenario background, and `person` should own it.

Reply:
Good point, added that